versionutils/version.go

```diff
@@ -125,17 +130,26 @@ func (v Version) IsGreaterThan(lesser Version) (bool, bool) {
 		return false, true
 	}
 
-	if v.Label != lesser.Label {
+	if v.Label == lesser.Label {
+		if v.LabelVersion > lesser.LabelVersion {
+			return true, true
+		} else if v.LabelVersion < lesser.LabelVersion {
+			return false, true
+		}
 		return false, false
 	}
 
-	if v.LabelVersion > lesser.LabelVersion {
-		return true, true
-	} else if v.LabelVersion < lesser.LabelVersion {
-		return false, true
+	// impose additional ordering based on our special labels
+	for _, label := range SpeciallyOrderedPrefixes {
+		if strings.HasPrefix(v.Label, label) {
```

Comment:
Oh also I would probably lean toward exact match here. I think that would work and save us from complex regex code you mentioned in the PR description, unless I'm missing something.